sql/dd/impl/upgrade/dd.cc

```diff
@@ -1031,6 +1032,124 @@ bool update_object_ids(THD *thd, const std::set<String_type> &create_set,
+    build_table_filename(new_table, sizeof(new_table) - 1,
+                         new_schema_name.c_str(), table_name.c_str(), "", 0);
+    err = handler->ha_rename_table(old_table, new_table, nullptr, nullptr);
+    if (err) {
+      char errbuf[MYSYS_STRERROR_SIZE];
+      LogErr(ERROR_LEVEL, ER_ERROR_ON_RENAME, old_table, new_table, err,
+             my_strerror(errbuf, sizeof(errbuf), err));
+      break;
+    }
+  }
+
+  destroy(handler);
+  return err;
+}
+
+/*
+  upgrade dd_properties table during dd_engine change
+
+  @param  thd                         Thread context.
+  @param  mysql_schema_id             Id of the 'mysql' schema.
+  @param  target_table_schema_id      Id of the 'target_table' schema
+                                      during upgrade, DD table target(expect)
+                                      version is created on target_table_schema
+  @param  target_table_schema_name    Name of the 'target_table_schema' schema
+  @param  actual_table_schema_id      Id of the 'actual_table' schema
+                                      during upgrade, DD table actual(old)
+                                      version is moved to 'actual_table' scheme
+
+  @returns false if success, true otherwise.
+
+*/
+bool upgrade_dd_properties_table(THD *thd, Object_id mysql_schema_id,
+                                 Object_id target_table_schema_id,
+                                 const String_type &target_table_schema_name,
+                                 Object_id actual_table_schema_id) {
+  auto dd_property_table_name = dd::tables::DD_properties::instance().name();
+  // only upgrade dd_properties table for dd_engine table
+  // during dd_version change, dd_properties table keep same
+  if (!bootstrap::DD_bootstrap_ctx::instance().is_dd_engine_change())
+    return false;
+
+  // create <target_table>.dd_properties table with default_ddse
+  auto *dd_properties_def =
+      dd::tables::DD_properties::instance().target_table_definition();
+  if (dd::execute_query(thd, dd_properties_def->get_ddl())) {
+    assert(false);
+    return dd::end_transaction(thd, true);
+  }
+
+  // copy data from existing mysql.dd_properties to <target_table>.dd_properties
+  std::stringstream ss;
+  ss << "INSERT INTO dd_properties SELECT * FROM " << MYSQL_SCHEMA_NAME.str
+     << "." << dd_property_table_name;
+  if (dd::execute_query(thd, ss.str().c_str())) {
+    assert(false);
+    return dd::end_transaction(thd, true);
+  }
+
+  // 'rename' <target_table>.dd_properties to mysql.dd_properties
+  std::set<String_type> dd_properties_set({dd_property_table_name});
+  if (update_object_ids(thd, dd_properties_set, dd_properties_set,
+                        mysql_schema_id, target_table_schema_id,
+                        target_table_schema_name, actual_table_schema_id)) {
+    assert(false);
+    return true;
+  }
+  dd::end_transaction(thd, false);
+
+  // myrocks has its own data dictionary which contains table name, call
```

Comment:
Redundant comment now that `update_myrocks_table_name` exists and has its own table header comment. Maybe add a bit more info there if needed